packages/core/src/test/shared/wizards/prompterTester.ts

```diff
@@ -24,38 +24,87 @@ export class PrompterTester {
 
     handleQuickPick(titlePattern: string, handler: (input: TestQuickPick) => void): PrompterTester {
         this.quickPickHandlers.set(titlePattern, handler)
+        this.handlerCallHistory.set(titlePattern, 0)
         return this
     }
 
     handleInputBox(titlePattern: string, handler: (input: TestInputBox) => void): PrompterTester {
         this.inputBoxHanlder.set(titlePattern, handler)
+        this.handlerCallHistory.set(titlePattern, 0)
         return this
     }
 
-    build(): void {
+    build(): PrompterTester {
         this.testWindow.onDidShowQuickPick((input) => {
             return this.handle(input, this.quickPickHandlers)
         })
         this.testWindow.onDidShowInputBox((input) => {
             return this.handle(input, this.inputBoxHanlder)
         })
+        return this
     }
 
     private record(title: string): void {
-        this.calledOrder++
-        this.report.set(title, this.calledOrder)
+        this.callLog.push(title)
+        this.handlerCallHistory.set(title, (this.handlerCallHistory.get(title) ?? 0) + 1)
+    }
+
+    /**
+     * Asserts that a specific handler has been called the expected number of times.
+     *
+     * @param title - The title or identifier of the handler to check.
+     * @param expectedCall - The expected number of times the handler should have been called.
+     * @throws AssertionError if the actual number of calls doesn't match the expected number.
+     */
+    assertHandlerCall(title: string, expectedCall: number) {
+        assert.strictEqual(this.handlerCallHistory.get(title), expectedCall, title)
+    }
+
+    /**
+     * Asserts that a specific handler was called in the expected order.
+     *
+     * @param title - The title or identifier of the handler to check.
+     * @param expectedOrder - The expected position in the call order (one-based index).
+     * @throws AssertionError if the handler wasn't called in the expected order.
+     */
+    assertCallOrder(title: string, expectedOrder: number) {
+        assert.strictEqual(this.callLog[expectedOrder - 1], title)
     }
 
-    public assertOrder(title: string, expectedOrder: number) {
-        assert.strictEqual(this.report.get(title) || 0, expectedOrder)
+    /**
+     * Asserts that all handler was called in the expected number of times.
```

Comment:
If 1 handler is called twice but the others only once, this function won't be useful?

```suggestion
     * Asserts that all handlers were called an expected number of times.
```

Reply:
You are right. I update the function to: 
```
    assertCallAll(titles: string[], expectedOrder: number) {
        titles.every((handler) => {
            this.assertCall(handler, expectedOrder)
        })
    }
```